test/unit.js

```diff
@@ -1036,6 +1034,252 @@ describe('Populator with specified unique tracked entity attribute ID with multi
   });
 });
 
+describe('Populator with specified unique data element ID', function() {
+  describe('#typeCache', function() {
+    it('should set specified uniqueDataElementID to the cache', function (next) {
+      var populatorOptions =
+      {
+        uniqueDataElementID: 'data element id'
+      };
+      var populator = new Populator(populatorOptions);
+      expect(populator._cache.uniqueEventDataElementID).to.equal('data element id');
+      expect(populator._options.duplicateThreshold).to.equal(-1)
+      next();
+    })
+
+    it('should not set uniqueTrackedEntityAttributeID to cache if option not passed in', function (next) {
+      var populatorOptions = {};
+      var populator = new Populator(populatorOptions);
+      expect(populator._cache.uniqueEventDataElementID).to.not.exist;
+      expect(populator._options.duplicateThreshold).to.equal(-1)
+      next();
+    });
+  });
+
+  describe('#checkForDuplicateEventUsingDataElementID', function() {
+    var requestMock;
+    var sandbox = Sinon.sandbox.create();
+    beforeEach(function (next) {
+      requestMock = sandbox.mock(Request);
+      next();
+    });
+
+    afterEach(function (next) {
+      sandbox.restore();
+      next();
+    });
+
+    it('should return error when DHIS2 sends non-200 response', function (next) {
+      var uniqueDataElementID = 'uniqueID123';
+      var eventUUID = '04ab1f15-f8c7-4cf3-9b80-8a01bd9a92cd';
+
+      var populatorOptions = {
+        uniqueDataElementID,
+        url: 'http://localhost/',
+        programID: 'some program id',
+        stageID: 'some stage id',
+        trackedEntityID: 'some tracked entity id'
+      };
+      var populator = new Populator(populatorOptions);
+
+      var response = {
+        statusCode: 500
+      };
+
+      var requestObject = {
+        method: 'GET',
+        path: '/api/events',
+      };
+
+      var expectedRequestObject = Sinon.match({
+        method: requestObject.method,
+        path: requestObject.path,
+        timestamp: Sinon.match.number
```

Comment:
Would it not be better to make use of the `Date` matcher instead? Unless this isnt an actual timestamp value?
https://sinonjs.org/releases/latest/matchers/#sinonmatchdate

Reply:
It's a number :| It's Date.now() which is milliseconds...